impl/src/main/java/org/ehcache/impl/internal/spi/copy/DefaultCopyProvider.java

```diff
@@ -54,6 +54,13 @@ public DefaultCopyProvider(DefaultCopyProviderConfiguration configuration) {
     return createCopier(DefaultCopierConfiguration.Type.VALUE, clazz, serializer, configs);
   }
 
+  @Override
+  public void releaseCopier(Copier<?> copier) throws Exception {
+    if (!(copier instanceof IdentityCopier) & !(copier instanceof SerializingCopier)) {
```

Comment:
Not sure I follow 100% your thought path. Can you check what happens if you _explicitly_ configure an `IdentityCopier` or a `SerializingCopier`? Cause I am pretty sure you leak references by leaving them in the `ClassInstanceProvider` instantiated set.